main.py

```diff
@@ -27,42 +31,105 @@ def setup(self):
 
         So, if you have any setup routine, insert it here.
         """
-        log.info("Setup from of_multi_table")
+        self.pipeline_controller = self.get_pipeline_controller()
 
     def execute(self):
         """Execute once when the napp is running."""
         log.info("Executing of_multi_table")
 
+    @staticmethod
+    def get_pipeline_controller():
+        """Get PipelineController"""
+        return PipelineController()
+
     @rest("/v1/pipeline", methods=["POST"])
     @validate_openapi(spec)
     def add_pipeline(self, data):
         """Add pipeline"""
-        return jsonify("Operation successful"), 201
+        log.debug("add_pipeline /v1/pipeline")
+        try:
+            _id = self.pipeline_controller.insert_pipeline(data)
+        except ValidationError as err:
+            msg = self.error_msg(err.errors())
+            log.debug("add_pipeline result %s %s", msg, 400)
+            raise BadRequest(msg) from err
+        msg = f"Pipeline {_id} created"
+        log.debug("add_pipeline result %s %s", _id, 201)
+        return jsonify(msg), 201
 
     @rest("/v1/pipeline", methods=["GET"])
     def list_pipelines(self):
         """List pipelines"""
-        return jsonify("Operation successful"), 201
+        log.debug("list_pipelines /v1/pipeline")
+        args = request.args.to_dict()
+        status = args.get("status", None)
+        pipelines = self.pipeline_controller.get_pipelines(status)
+        log.debug("list_pipelines result %s %s", pipelines, 201)
+        return jsonify(pipelines), 201
```

Comment:
This should be returning 200, also `log.debug` for listing isn't that much valuable, can we get rid or them? After all, this method here either it'll execute and return or crash, also logging all the pipelines can get very noisy even though it's debug level on e2e this will be sent to logs.

Reply:
Got it, deleted `log.debug` with dictionaries in them.